app/src/main/java/com/example/bigowlapp/fragments/SupervisionResponseFragment.java

```diff
@@ -0,0 +1,96 @@
+package com.example.bigowlapp.fragments;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.LiveData;
+
+import com.example.bigowlapp.R;
+import com.example.bigowlapp.model.Group;
+import com.example.bigowlapp.model.SupervisionRequest;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.repository.RepositoryFacade;
+
+public class SupervisionResponseFragment extends Fragment {
+
+    private Button acceptBtn;
+    private Button rejectBtn;
+    private TextView groupName;
+    private TextView groupSupervisor;
+    private RepositoryFacade repositoryFacade;
+    private LiveData<Group> groupLiveData;
+    private LiveData<User> userLiveData;
+    SupervisionRequest supervisionRequest;
+
+    public SupervisionResponseFragment() {
+
+    }
+
+    public SupervisionResponseFragment(SupervisionRequest sr) {
+        this.supervisionRequest = sr;
+    }
+
+    public static SupervisionResponseFragment newInstance(SupervisionRequest sr) {
+        SupervisionResponseFragment fragment = new SupervisionResponseFragment(sr);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
```

Comment:
Try to break `onCreateView` down a bit to make it a bit more readable, and not just one huge function that does everything..

you can extract sections of the method into new methods. This would also be an `Extract Method` refactor.

You can do this really easily by highlighting the section of code you want to extract, right click in android studio, hover on Refactor, then click Extract Method. I put an example of what I mean below.

![image](https://user-images.githubusercontent.com/31963426/109754843-c84fcc00-7bb2-11eb-88e1-d9a1ed0c80c4.png)